src/bin/cargo/main.rs

```diff
@@ -168,7 +168,13 @@ fn execute_external_subcommand(config: &Config, cmd: &str, args: &[&str]) -> Cli
         None => {
             let suggestions = list_commands(config);
             let did_you_mean = closest_msg(cmd, suggestions.keys(), |c| c);
-            let err = anyhow::format_err!("no such subcommand: `{}`{}", cmd, did_you_mean);
+
+            let err = if cmd.contains('+') {
+                anyhow::format_err!("no such subcommand: `{}`{}\n\n\tCargo does not handle `+toolchain` directives itself.\n\tDid you mean to run `cargo` through `rustup` instead?\n\tsuggestion: view a list of all installed Cargo subcommands using `cargo --list`", cmd, did_you_mean)
```

Comment:
Suggesting to run `cargo --list` here does not make sense as this is about using `+nightly` or similar

Reply:
Indeed. Thank you for pointing this out.